### ISSUE TITLE
<webview> not rendering content after reload or redirect (Electron 5.x, 6.x)

### ISSUE THREAD

**Issue by Catlinz** [electron/electron#18177]
### Preflight Checklist
* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 5.x & 6.0.0-beta.1
* **Operating System:**
  *  Windows 10 (1803)
* **Last Known Working Electron version:**:
  * 4.x

### Expected Behavior
When creating a webview with `src="http://mail.google.com`", I expect it to load and show the google login page.  When reloading the webview (or if the webview redirects e.g., when logging in) I expect it to also properly render the page.

### Actual Behavior
Sometimes when loading the above URL, nothing is visible.  Always when reloading the webview with the above URL, the webview will no longer render the page.  E.g., signing into things like gmail, isn't really working because it signs you in, but you end up with a non-rendering webview.

### To Reproduce
Replace the renderer process script in electron quickstart / electron fiddle with the following, and click the refresh button (with `webviewTag: true` in webPreferences).
```javascript
document.addEventListener('DOMContentLoaded', () => {
    const wv = document.createElement('webview');
    wv.src = 'https://mail.google.com';
    wv.style.height = '400px';

    const a = document.createElement('a');
    a.innerText = 'REFRESH';
    a.addEventListener('click', () => wv.reload());

    document.body.append(wv, a);
});
```

### Screenshots
What it renders (usually) when first loaded (and what should be rendered):
![what_should_render](https://user-images.githubusercontent.com/11395871/57236898-e100f200-6fda-11e9-9fe6-d99bc4b482b5.png)

What renders after the "REFRESH" button is clicked:
![after_refresh](https://user-images.githubusercontent.com/11395871/57236940-f249fe80-6fda-11e9-8b1d-75963bf79f92.png)

### Additional Information
The webview (and iframe within it) still have their proper sizes.  The page within the webview is actually loading correctly, if you inspect it you can see all the page is there.

If you change CSS properties in the dev tools that cause the webview to be re-rendered, the page then renders correctly.  (e.g., after reloading and getting the blank page, if you toggle the `display: flex;` CSS property on the webview, the page will render.  Just resizing the window that the webview is in does not cause the page to render.

It seems to be affecting all Electron versions from 5.0.0-beta.1 and onward.

**Comment by 4eveRS** [comment]
Found the same behavior on my application. Looking for a fix.

**Comment by littlecold233** [comment]
same problem. Looking for a fix

**Comment by littlecold233** [comment]
It doesn't work in MacOS either

**Comment by shockthetoast** [comment]
We just updated to 5.0.1 and our webview no longer renders, either. However, adjusting css doesn't seem to restore it for us. Additionally, the webview's "dom-ready" event never fires.

**Comment by javan** [comment]
I'm seeing this issue too, and I believe it's specific to `<webview>`s  containing `<iframe>`s. Possibly related: https://github.com/electron/electron/issues/17890, https://github.com/electron/electron/issues/18019.

**Comment by codebytere** [comment]
@javan are you seeing this with the above repro on 5 and 6? i'm not able to reproduce it using quick start and the above repro.

**Comment by javan** [comment]
@codebytere, I haven't tried the above repro, but the issue happening in our (Basecamp) app and preventing us from upgrading to Electron v5. 

I managed to narrow down reproduction steps to a fairly minimal test case. I suspect the problem occurs when dynamically created `<iframe>`s are added and/or removed from the DOM (Basecamp does this to show embedded URLs like tweets, youtube videos, etc.).

To reproduce, run: https://github.com/javan/electron-quick-start/tree/webview-iframe-bug

1. `$ git clone https://github.com/javan/electron-quick-start.git -b webview-iframe-bug`
1. `$ npm install`
1. `$ npm start`
1. Click the "Embed 1" button
1. Click the "Embed 2" button
1. Click the "Embed 1" button again

At this point, the `<webview>` seems to freeze. The dynamically inserted content doesn't appear and the button stays highlighted. Also, the app stops responding to keyboard commands like <kbd>⌘</kbd>+<kbd>W</kbd> and <kbd>⌘</kbd>+<kbd>Q</kbd>.

<details>
<summary>Demo</summary>

![webview-iframe](https://user-images.githubusercontent.com/5355/58190481-9940c280-7c8a-11e9-8143-8a7c187d95e6.gif)

</details>

**Comment by javan** [comment]
Note: The above example doesn't reload or redirect the `<webview>` (like the title of this issue mentions). It's a single page that adds and removes `<iframe>`s from the DOM.

**Comment by Catlinz** [comment]
So, I just did some checking, and this issue _appears_ to have been fixed in v6.0.0-beta.2.  Not entirely sure what fixed it, but I don't seem to be getting the original issue when running the code on Electron Fiddle.

**Comment by gapsong** [comment]
@Catlinz could you please close this issue, if it has been fixed already?
